### PR TITLE
Milestone 3: Setup association

### PR DESCRIPTION
**Pull request by AngomRanjan**
#### Changes made in this milestone:

- Created a class `Classroom` 
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`
- Created a class `Book` 
- Created a class `Rental`
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rentals.book = self
+  end
```
Comment (brytebee) [inline book.rb]: - [ ] **Required:** Kindly refactor the method such that the input to the add_rental method would be a `person` and the `date` in which the rental is added.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @AngomRanjan👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️

- Kindly update the Person class. Include a add_rental method, a book is rented on a given date.
https://github.com/AngomRanjan/oop-school-library/blob/0efb742f068b90776487da018973d251049143eb/person.rb#L3

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AngomRanjan** [comment]
## Made changes as suggested

- Included add_rental Method in Person Class.

File: book.rb (hunk 1 above)
Comment (AngomRanjan) [reply book.rb]: Refactored the add_rental method as suggested

**Comment by AngomRanjan** [comment]
## Refactored the add_rental method as suggested

**Pull request merged**